spec/src/provider.spec.js

```diff
@@ -101,6 +101,93 @@ describe("Provider", function() {
     });
   });
 
+  it("rejects method calls when the provider errors", function (done) {
+    var iface = port.getInterface();
+    var o = function() {};
+    o.prototype.m1 = function(str) {
+      throw new Error('hi.myerr');
+    };
+
+    iface.providePromises(o);
+    port.onMessage('default', {
+      channel: 'message'
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message:{
+      'type': 'construct',
+    }});
+
+    port.on('message', function(n) {
+      expect(n.message.error).toContain('hi.myerr');
+      done();
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message: {
+      'action': 'method',
+      'type': 'm1',
+      'text': ['mystr'],
+      'reqId': 1
+    }});
+  });
+
+  iit("rejects method calls when the provider doesn't return a promise", function (done) {
+    var iface = port.getInterface();
+    var o = function() {};
+    o.prototype.m1 = function(str) {
+      return 'a weird return';
+    };
+
+    iface.providePromises(o);
+    port.onMessage('default', {
+      channel: 'message'
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message:{
+      'type': 'construct',
+    }});
+
+    port.on('message', function(n) {
+      expect(n.message.error).toContain('a weird return');
+      done();
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message: {
+      'action': 'method',
+      'type': 'm1',
+      'text': ['mystr'],
+      'reqId': 1
+    }});
+  });
+
+  it("rejects method calls when the provider doesn't exist", function (done) {
+    var iface = port.getInterface();
+    var o = function() {};
+    var called = false, resp;
+
+    iface.providePromises(o);
+    port.onMessage('default', {
+      channel: 'message'
+    });
+
+    port.onMessage('default', {to: 'testInst', type:'message', message:{
+      'type': 'construct',
+    }});
+
+    port.onMessage('default', {to: 'testInst', type:'message', message: {
+      'action': 'method',
+      'type': 'm1',
+      'text': ['mystr'],
+      'reqId': 1
+    }});
+
+    expect(called).toEqual(true);
+
+    port.on('message', function(n) {
+      expect(n.message.error.length).toBeGreaterThan(0);
```

Comment:
What is the behavior when n.message.error contains more than 1 error? How does the Promise.reject know what to return?